app/models/spree/credit_card_decorator.rb

```diff
@@ -31,13 +31,17 @@ def save_requested_by_customer?
 
   private
 
+  def reusable?
+    gateway_customer_profile_id.present?
+  end
+
   def default_missing?
     !user.credit_cards.exists?(is_default: true)
   end
 
   def ensure_single_default_card
     return unless user
-    return unless is_default? || default_missing?
+    return unless is_default? || (reusable? && default_missing?)
```

Comment:
this condition is so short and still so brain damaging :-)
I am not sure how to make it better. extract method probably.